packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -1085,6 +1085,48 @@ mixin SchedulerBinding on BindingBase {
     }
   }
 
+  final Map<dynamic, DartPerformanceMode> _performanceModes = <Object, DartPerformanceMode>{};
+
+  /// Request a specific [DartPerformanceMode]. `requestor` is the handle to
+  /// component making the request, typically the component itself. Returns
+  /// `true` is the request was successfully made to the engine, `false` otherwise.
+  /// Even if the result is `true`, the engine may choose to ignore the request or
+  /// the performance mode may not be guaranteed to be the one requested.
+  ///
+  /// If conflicting requests are made, only the first request will be honored.
+  bool createPerformanceModeRequest(Object requestor, DartPerformanceMode mode) {
+    if (_performanceModes.isNotEmpty) {
+      final DartPerformanceMode oldRequest = _performanceModes.entries.first.value;
+      if (oldRequest != mode) {
+        return false;
+      }
+    }
+    _performanceModes[requestor] = mode;
+    PlatformDispatcher.instance.requestDartPerformanceMode(mode);
+    return true;
+  }
+
+  /// Remove a request for a specific [DartPerformanceMode]. `requestor` is the
+  /// handle to component making the request, typically the component itself. If
+  /// all the pending requests have been disposed, the engine will revert to the
+  /// [DartPerformanceMode.balanced] performance mode.
+  void disposePerformanceModeRequest(Object requestor) {
+    _performanceModes.remove(requestor);
```

Comment:
if this removes the first mode in the list, shouldn't we activate the next mode in it?